### PR TITLE
Upgrade CI not to use deprecated set-env command

### LINKED ISSUE #821: Upgrade CI not to use deprecated set-env command
https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

### PR DESCRIPTION
**Pull request by michallepicki**
Closes #821

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -69,12 +69,12 @@ jobs:
...
             cp "target/${{ matrix.target }}/release/gleam.exe" "gleam.exe"
             7z a "$ARCHIVE" "gleam.exe"
-            echo "::set-env name=ASSET::$ARCHIVE"
+            echo "name=ASSET::$ARCHIVE" >> $GITHUB_ENV
           else
             ARCHIVE="gleam-$VERSION-${{ matrix.build }}.tar.gz"
             cp "target/${{ matrix.target }}/release/gleam" "gleam"
             tar -czvf "$ARCHIVE" "gleam"
-            echo "::set-env name=ASSET::$ARCHIVE"
+            echo "name=ASSET::$ARCHIVE" >> $GITHUB_ENV
```
Comment (lpil) [inline .github/workflows/release.yaml:77]: Is this right? I can't see the `::` in the docs here?

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files

Comment (michallepicki) [reply .github/workflows/release.yaml:77]: Silly me, I misunderstood how the previous thing worked (never had a chance to use it and it looks like they provide only new documentation now). I'll fix it soon.

Comment (michallepicki) [reply .github/workflows/release.yaml]: Updated, this should do it.

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**